lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/client/YarnSubmissionHelper.java

```diff
@@ -279,6 +282,13 @@ public void submit() throws IOException, YarnException {
     }
 
     this.yarnClient.submitApplication(applicationSubmissionContext);
+
+    if (this.isUnmanaged) {
+      // For Unmanaged AM mode, add a new app token to the
+      // current process so it can talk to the RM as an AM.
+      final Token<AMRMTokenIdentifier> token = this.yarnClient.getAMRMToken(this.applicationId);
```

Comment:
@jwang98052 no, that's a new token that the Driver gets from the RM. Since in Unmanaged AM mode the Driver runs in the Client process, we need to add that token to the current process' credentials - otherwise, the Driver won't be able to talk to YARN.